src/aioftp/pathio.py

```diff
@@ -482,217 +617,274 @@ class AsyncPathIO(AbstractPathIO):
     :type executor: :py:class:`concurrent.futures.Executor`
     """
 
-    def __init__(self, *args, executor=None, **kwargs):
-        super().__init__(*args, **kwargs)
+    executor: Union[Executor, None]
```

Comment:
Hm, do we need this here, on class level? Is `self.executor: Union[Executor, None] = executor` from `__init__` not enough?